### ISSUE TITLE
Machine Learning 輪講

### ISSUE THREAD

**Issue by Hayashi-Yudai** [wantedly/machine-learning-round-table#334]
## Why

Machine Learning 輪講は最新の技術や論文を追うことで、エンジニアが「技術で解決できること」のレベルをあげていくことを目的にした会です。

prev. https://github.com/wantedly/machine-learning-round-table/issues/331

## What

話したいことがある人はここにコメントしましょう！
面白いものを見つけた時点でとりあえず話すという宣言だけでもしましょう！

**Comment by ghibney** [comment]
## [Diversification as Risk Minimization](https://dl.acm.org/doi/10.1145/3773966.3777985)

### 著者
Rikiya Takehi, Fernando Diaz, Tetsuya Sakai

### 学会
WSDM '26 の Best Full Paper

### どんな論文
* 複数の意図が考えられるクエリ（例：jaguar）では、多数派の意図に最適化されやすく、少数派ユーザが強い失敗体験をするという問題がある
* 既存の手法では、最悪ケース（少数意図）を守る保証がない
* 多様化をリスク最小化として再定式化し、平均性能の毀損を抑えながら最も満たされていない意図の損失（worst-case loss）を改善する手法を提案している
### 背景
* 従来の検索ロバスト性研究はクエリ間のリスク最小化（少数派クエリを救うような手法）を対象とし、「クエリ内の意図間リスク」は扱っていない。

### 手法
* 各意図に対し目標値との差分をヒンジ損失で定義し、その損失分布の最悪β割合の期待値として VRisk（CVaR） を定義している

<img width="391" height="312" alt="Image" src="https://github.com/user-attachments/assets/895b3820-3e8a-4bdd-818b-9fe87d3638a4" />

### 結果
* NTCIR INTENT-2、TREC Web 2012、MovieLens 32Mで評価
* VRiskerは worst-case loss を20–33%削減しつつ、平均性能の低下は約2%前後に抑えられている

<img width="509" height="351" alt="Image" src="https://github.com/user-attachments/assets/84dfef4b-7bd1-4b29-983f-846e4c96c72a" />

<img width="563" height="287" alt="Image" src="https://github.com/user-attachments/assets/90ad923b-bbd3-42ad-9852-6a192614725c" />

**Comment by Hayashi-Yudai** [comment]
## [Recommendation as Language Processing (RLP): A Unified Pretrain, Personalized Prompt & Predict Paradigm (P5)](https://www.alphaxiv.org/abs/2203.13366)

T5 を生成的推薦タスク用に拡張したモデル。

<img width="748" height="409" alt="Image" src="https://github.com/user-attachments/assets/eb896d9c-c65f-4bc6-970e-ec7f7139b1a5" />

- Token Emb.: 通常の SentencePiece を使ってトークナイズして Embedding 化したもの
- Position Emb.: 位置エンコーディング
- Whole-word Emb.: 自然言語レベルでの単語の Embedding。"user_123" などは SentencePiece を使うと分割されてしまうので、それを防いだもの

の3種類の Embedding を利用している。また、各タスクについて複数のプロンプトテンプレートを用意することで、大量の学習データを確保。

## [OpenP5: An Open-Source Platform for Developing, Training, and Evaluating LLM-based Recommender Systems](https://www.alphaxiv.org/abs/2306.11134)

P5 の再現実装、研究用のプラットフォーム。P5 とは異なる点がいくつかある

- オリジナルとは異なり 2つの推薦タスクでのみ学習が行われている
- バックボーンモデルとして T5 の他に Llama が使えるようになっている (性能としては T5 の方が上)

## [RDRec: Rationale Distillation for LLM-based Recommendation](https://www.alphaxiv.org/abs/2405.10587)

アイデアとしては P5 と似ている。学習するタスクの設計が P5 と違う(と思っている)

<img width="762" height="255" alt="Image" src="https://github.com/user-attachments/assets/cc9da426-2d6d-4e9c-981d-92d83b2f970e" />

ユーザーのアイテムに対するレビューからユーザーがアイテムを好む理由を LLM によって抽出し、それを正解データとして提案モデルの学習に利用している。

**Comment by chimuichimu** [comment]
## [On the Reliability of Sampling Strategies in Offline Recommender Evaluation](https://dl.acm.org/doi/10.1145/3705328.3748086)

### Summary

推薦システムのオフライン評価において、exposure bias とサンプリングバイアスが評価結果に与える影響を体系的に分析した研究。完全観測データセット（KuaiRec）を ground truth として用い、様々な exposure bias 条件下で9種類のサンプリング戦略の信頼性を4つの次元（resolution、fidelity、robustness、predictive power）から評価した。サンプリング戦略の選択が評価結果を大きく歪め得ることを示し、実務的なガイダンスを提供している。

### Journal / Conference

RecSys 2025

### Background

推薦システムのオフライン評価は、オンライン A/B テストが高コスト・高リスクであるため広く用いられているが、2つのバイアスに影響を受ける。1つ目は exposure bias（ユーザーは表示されたアイテムとしかインタラクションできない）、2つ目は sampling bias（計算コスト削減のためにネガティブアイテムをサンプリングして top-k 指標を近似する際に生じる歪み）である。

### Motivation

- 既存研究はサンプリング戦略を固定されたログデータ上でのみ評価しており、異なる exposure bias 条件下での信頼性や、真のユーザー嗜好に対する一致度は検証されていなかった
- ログデータ自体を ground truth として扱う研究が多く、サンプリング評価が実際のユーザー嗜好をどの程度反映するかは未解明だった
- exposure bias とサンプリングバイアスの効果を分離して分析する枠組みが必要だった

### Method

- **概念的枠組み**: オフライン評価を3段階に分離して定式化
  - Ground-Truth Preferences（G）: 完全観測されたユーザー・アイテム嗜好行列
  - Logged Interactions（L）: exposure policy を適用して部分観測された行列
  - Sampled Interactions（G_S, L_S）: G or L にサンプリング戦略を適用した評価用行列
- **評価軸**:
  - Resolution: サンプラーがモデル間の性能差を識別できるか
  - Fidelity: サンプリング評価がフル評価と一致するか（L_S vs L、G_S vs G）
  - Robustness: バイアス条件下で評価結果が安定するか（L_S vs G_S）
  - Predictive Power: バイアスのあるサンプルから ground truth のモデルランキングを復元できるか（L_S vs G）
- **実験設定**:
  - データセット: KuaiRec（テスト分割の密度が約100%で完全観測に近い）
  - ロギング戦略: Uniform / Popularity-biased / Positivity-biased（各8段階のスパース度）
  - サンプリング戦略: Full, Exposed, Random@e, Random@n, Popularity@n, Positivity@n, WTD@n, WTDH@n, Skew@n の9種類
  - 推薦モデル: ALS, BPR, LightFM（2変種）, SAR-Cosine, SAR-Jaccard, Popularity, Random の7モデル
  - 合計1,512の評価シナリオ

### Result

評価指標は主に nDCG@100、メタ評価指標として tie rate（Resolution）と Kendall's τ（その他3次元）を使用。

- **Resolution（Q1）**: サンプル数が極端に少ない or 多いとタイが増加し、小〜中程度が最良。高スパース条件では Skew、Popularity、Positivity が比較的良好な識別力を維持
- **Fidelity（Q2）**: WTD、WTDH、Random が n ≥ 200 で高い fidelity に到達。Popularity と Positivity は収束が遅い。0%スパース条件でもサンプラー間で差があり、サンプリング自体が exposure bias とは独立に歪みを生む
- **Robustness（Q3）**: WTD、WTDH が exposure bias に対して最も頑健。Exposed はロガーバイアスの影響を強く受け、同数の Random@e に劣る
- **Predictive Power（Q4）**: Skew、WTD、WTDH が総合的に最良で、10%〜50%スパースで n ≥ 200 あれば Kendall's τ > 0.7。Random もこれらに匹敵し、シンプルかつ有効な選択肢

本研究により得られた知見

- 全次元で最良な単一サンプラーは存在せず、トレードオフがある
- サンプル数だけでなくサンプル構成（どのアイテムを選ぶか）が重要
- バイアス考慮手法（WTD、WTDH、Skew）が総合的に優れる
- 完全観測データセットの希少性が一般化の課題

**Comment by nogawanogawa** [comment]
# Augmenting graph convolutional networks with textual data for recommendations
## 論文URL

https://www.amazon.science/publications/augmenting-graph-convolutional-networks-with-textual-data-for-recommendations

## 著者

Sergey Volokhin, Marcus Collins, Oleg Rokhlenko, Eugene Agichtein

## 会議

ECIR'23 

## 背景

GNNを推薦モデルに利用する研究が進められている。
ユーザーレビューやアイテムの説明文といった形式で膨大な量の非構造化テキストが利用可能であるにもかかわらず、その多くはアイテムーユーザー間の相互作用だけを対象にしている。

## 目的

- GNNでサイド情報を利用できるように拡張する

## アプローチ

- TextGCN: GNN推薦システムに非構造化テキストを組み込む
  - LightGCN へテキストを組み込む拡張でなるべくシンプルな構造

<img width="736" height="548" alt="Image" src="https://github.com/user-attachments/assets/23318f1e-cd5a-4863-a656-3872a563c52f" />

## memo

- https://github.com/nogawanogawa/paper_memo/issues/154
